crates/cdk/src/wallet/mod.rs

```diff
@@ -500,17 +497,35 @@ impl Wallet {
     ///     let wallet = Wallet::new(mint_url, unit, Arc::new(localstore), &seed, None)?;
     ///     let amount = Amount::from(100);
     ///
-    ///     let quote = wallet.mint_quote(amount).await?;
+    ///     let quote = wallet.mint_quote(amount, None).await?;
     ///     Ok(())
     /// }
     /// ```
     #[instrument(skip(self))]
-    pub async fn mint_quote(&self, amount: Amount) -> Result<MintQuote, Error> {
+    pub async fn mint_quote(&self, amount: Amount, description: Option<String>) -> Result<MintQuote, Error> {
         let mint_url = self.mint_url.clone();
         let unit = self.unit;
+
+        // If we have a description, we check that the mint supports it.
+        if description.is_some() {
+            if let Some(mint_method_settings) = self.localstore
+                .get_mint(mint_url.clone())
+                .await?
+                .expect("Could not retrieve mint information")
+                .nuts
+                .nut04
+                .get_settings(&unit, &PaymentMethod::Bolt11) {
+                if !mint_method_settings.description {
+                    return Err(Error::InvoiceDescriptionUnsupported);
+                }
+            } else {
+                return Err(Error::UnitUnsupported);
+            }
+        }
```

Comment:
```suggestion
        // If we have a description, we check that the mint supports it.
        if description.is_some() {
            let mint_method_settings = self
                .localstore
                .get_mint(mint_url.clone())
                .await?
                .ok_or(Error::IncorrectMint)?
                .nuts
                .nut04
                .get_settings(&unit, &PaymentMethod::Bolt11)
                .ok_or(Error::UnsupportedUnit)?;

            if !mint_method_settings.description {
                return Err(Error::InvoiceDescriptionUnsupported);
            }
        }

```

We can do it without the match in this case since the error should just be bubbled up.

Reply:
Oh wow ok 😅